### PR TITLE
ringbuffer: add hint about tsrb to doc

### PR DESCRIPTION
**Pull request by kroesche**
This PR is all doc-only change.  In the first commit, I added some documentation to ringbuffer.h header to briefly describe the module.  I added a comment to refer the reader to tsrb in case they need a thread-safe ring buffer.

The second commit I slightly change doc of tsrb.  Previously the important information about usage restriction and buffer size did not show up on the API doc page but instead on the file doc page, which required an extra click to view.  It was easy to miss (I missed it the first time).  Also I added very brief phrase to api doc about buffer size that buffer size must be power of 2.  I think these changes make it less likely a user will make a usage mistake that will cost debug time.

### CODE REVIEW

**Comment by miri64** [comment]
@kroesche please limit the short description of your commits (the first line of your commit message) to 50 (72 at most) characters. For more detailed descriptions you can use the rest of the commit message.

**Comment by miri64** [comment]
> @kroesche please limit the short description of your commits (the first line of your commit message) to 50 (72 at most) characters. For more detailed descriptions you can use the rest of the commit message.

And please use something more descriptive than just "Doc-only change" ;-).

**Comment by kroesche** [comment]
Okay I will modify my commit messages as you suggest.  My purpose of adding "doc-only change" was not to replace the more detailed information that followed but to quickly inform the reader that there was no code change.  If that is not useful then I will leave it off.

Aside from that do you think this PR will be accepted?  I will close this PR and submit again.

**Comment by miri64** [comment]
> Okay I will modify my commit messages as you suggest. My purpose of adding "doc-only change" was not to replace the more detailed information that followed but to quickly inform the reader that there was no code change. If that is not useful then I will leave it off.

I usually mark doc-only changes by using a `doc: ` prefix in the commit message. Don't know if that is common practice however.

> Aside from that do you think this PR will be accepted? I will close this PR and submit again.

Just reword the commit messages for now, everything else can be done doing some fixups and (later) squashing during the review. For neither you need to open a PR (it would make it hard to follow discussions).

**Comment by kroesche** [comment]
I rewrote my commit messages and force-pushed my branch.  I hope thats what you meant I should do.  The new commit messages show up in the PR now.

**Comment by miri64** [comment]
> I rewrote my commit messages and force-pushed my branch. I hope thats what you meant I should do. The new commit messages show up in the PR now.

Yes :-)

File: sys/include/tsrb.h [hunk 1]
```diff
@@ -151,3 +151,4 @@ int tsrb_add(tsrb_t *rb, const char *src, size_t n);
 
 #endif /* TSRB_H */
 /** @} */
+
```
Comment (kaspar030) [inline sys/include/tsrb.h]: extra newline

**Comment by kaspar030** [comment]
ACK.

**Comment by kaspar030** [comment]
@kroesche please squash the last commit...

**Comment by kroesche** [comment]
I squashed the last commit into the second, leaving 2 commits in the PR.  If thats not what you meant let me know and I'll change it.

**Comment by kaspar030** [comment]
&go.

Thanks @kroesche!
